pymc/model.py

```diff
@@ -2035,10 +2035,43 @@ def Deterministic(name, var, model=None, dims=None):
 def Potential(name, var, model=None):
     """Add an arbitrary factor potential to the model likelihood
 
+      Potential function is used to add arbitrary factors i.e. constraints that we define to the likelihood function of a model to adjust the probablity density of the model. 
+
+    .. code:: python
+
+        with pm.Model() as my_model:
+          x = pm.Normal('x', mu=0, tau=1)
+          y = pm.Normal('y', mu=x, tau=1)
+
+          constraint = x + y < 1
+          potential = pm.Potential('constraint', constraint, my_model)
+
+        In this example, we define a constraint on the sum of x and y that it can only be lesser than 1. We pass this constraint to the pm.Potential function.
+        Now the pm.Potential function takes in a expression, a scalar value and the model name as argument.
+        Now, the probablity density that the model produces takes into consideration the constraint that the sum of x and y should be lesser than 1.
+
+        The Potential function here acts as a gatekeeper allowing only samples that satisfy the constraint to contribute to the likelihood. This ensures that the sample generated by the sampler are consistent with the constraint.
+
+        Have a look at the following example:
+       
+     .. code:: python
+
+        with pm.Model() as model:
+          x = pm.Normal('x', mu=0, sigma=1)
+          y = pm.Normal('y',mu=x, sigma=1,observed=data)
+          constraint = x>= 0
+          potential = pm.Potential('x_constraint',-tt.log(tt.switch(constraint, 1, 1e-30)))
```

Comment:
```suggestion
          potential = pm.Potential('x_constraint', -pt.log(pt.switch(constraint, 1, 1e-30)))
```

We recently switched our backend to [PyTensor](https://github.com/pymc-devs/pytensor)!

Reply:
If we are avoiding to use too many new concepts at once, can we use pm.math for backend computation here instead of explicitly mentioning Theano tensor or PyTensor?
I am quite new to PyTensor and am facing errors when I try to run the suggestion made above.

Reply:
Yes, that's a great idea. @ricardoV94 mentioned it below too